src/lib/seam/components/SupportedDeviceTable/SupportedDeviceContent.tsx

```diff
@@ -1,138 +1,111 @@
 import { Button } from 'lib/ui/Button.js'
 
-import { SupportedDeviceHeader } from './SupportedDeviceHeader.js'
-import { SupportedDeviceRow } from './SupportedDeviceRow.js'
-
-export interface SupportedDeviceContentProps {
-  cannotFilter?: boolean
+interface SupportedDeviceContentProps {
+  filterValue: string
+  resetFilterValue: () => void
+  filters: DeviceModelFilters
 }
 
 export function SupportedDeviceContent({
-  cannotFilter = false,
-}: SupportedDeviceContentProps): JSX.Element {
-  const [filterValue, setFilterValue] = useState('')
-  const [filters, setFilters] = useState<Filters>({
-    supportedOnly: false,
-    category: null,
-    brand: null,
-  })
-
-  const {
-    data: deviceModels,
-    isLoading,
-    isError,
-    refetch,
-  } = useQuery<DeviceModel[]>({
-    queryKey: ['supported_devices', filterValue, filters],
-    queryFn: async () => {
-      const url = new URL('https://devicedb.seam.co/api/device_models/list')
-
-      if (filterValue.trim() !== '') {
-        url.searchParams.set('text_search', filterValue.trim())
-      }
-
-      if (filters.supportedOnly) {
-        url.searchParams.set('support_level', 'live')
-      }
+  resetFilterValue,
+  filterValue,
+  filters,
+}: SupportedDeviceContentProps): JSX.Element | null {
+  const { deviceModels, isLoading, isError, refetch } = useFilteredDeviceModels(
+    filterValue,
+    filters
+  )
 
-      if (filters.category !== null) {
-        url.searchParams.set('main_category', filters.category)
-      }
+  if (isLoading) {
+    return (
+      <div className='seam-supported-device-table-content-state-block'>
+        <p>{t.loading}</p>
+      </div>
+    )
+  }
 
-      if (filters.brand !== null) {
-        url.searchParams.set('brand', filters.brand)
-      }
+  if (isError) {
+    return (
+      <div className='seam-supported-device-table-content-state-block'>
+        <p>{t.error}</p>
+        <Button
+          variant='solid'
+          size='small'
+          onClick={() => {
+            void refetch()
+          }}
+        >
+          {t.retry}
+        </Button>
+      </div>
+    )
+  }
 
-      const res = await fetch(url)
-      if (!res.ok) {
-        throw new Error('Failed to load device models')
-      }
-      const data = await res.json()
-      return data?.device_models ?? []
-    },
-  })
+  if (deviceModels == null) {
+    return null
+  }
 
   return (
-    <div className='seam-supported-device-table-content-wrap'>
-      {!cannotFilter && (
-        <SupportedDeviceFilterArea
-          deviceModels={deviceModels ?? []}
-          filterValue={filterValue}
-          setFilterValue={setFilterValue}
-          filters={filters}
-          setFilters={setFilters}
-        />
-      )}
+    <table className='seam-supported-device-table-content'>
+      <SupportedDeviceHeader />
+      <tbody>
+        {deviceModels.map((deviceModel) => (
+          <SupportedDeviceRow
+            key={deviceModel.manufacturer_model_id}
```

Comment:
Is `manufacturer_model_id` unique? Will we eventually have a proper unique id to use here?

Reply:
For now I'd probably do `key={`${index}:${deviceModel.manufacturer_model_id}`}` or something to that effect because IIRC I ran into a duplicate key issue. We could have the backend generate a UUID when it parses the CSV file as a more permanent solution.